c7n/filters/securityhub.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2017 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from c7n.utils import local_session, chunks, type_schema
+from .core import Filter
+from c7n.manager import resources
+
+
+class SecurityHubFindingFilter(Filter):
+    """Check if there are Security Hub Findings related to the resources
+    """
+    schema = type_schema('findings')
+    permissions = ('securityhub:GetFindings',)
+
+    def get_arn(self, r):
+        # need to make generic to support other resource types
+        return "arn:aws:{}:{}:instance/{}".format(
+            self.manager.config.region,
+            self.manager.config.account_id,
+            r["InstanceId"])
+
+    def process(self, resources, event=None):
+        if not resources:
```

Comment:
this shouldn't be nesc, actions aren't invoked with empty resource sets.

Reply:
nope, we short-circuit execution once we get to an empty set on both filters and actions as there isn't anything for them to process.

Reply:
ok, I've removed from securityhub code and will later submit to remove from line 41-42 of health.py which is the example I used as a starting point.